jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -479,20 +479,25 @@ def set_plot_axes(self):
 
         # Set axes labels for the spectrum viewer
         spectral_axis_unit_type = str(data.spectral_axis.unit.physical_type).title()
-        # flux_unit_type = data.flux.unit.physical_type.title()
+        spectral_axis_unit_str = data.spectral_axis.unit.to_string()
         flux_unit_type = "Flux density"
+        flux_unit_str = data.flux.unit.to_string()
 
         if data.spectral_axis.unit.is_equivalent(u.m):
             spectral_axis_unit_type = "Wavelength"
         elif data.spectral_axis.unit.is_equivalent(u.pixel):
             spectral_axis_unit_type = "pixel"
 
-        label_0 = f"{spectral_axis_unit_type} [{data.spectral_axis.unit.to_string()}]"
+        label_0 = f"{spectral_axis_unit_type} [{spectral_axis_unit_str}]"
         self.figure.axes[0].label = label_0
-        self.figure.axes[1].label = f"{flux_unit_type} [{data.flux.unit.to_string()}]"
+        self.figure.axes[1].label = f"{flux_unit_type} [{flux_unit_str}]"
 
         # Make it so y axis label is not covering tick numbers.
         self.figure.axes[1].label_offset = "-50"
 
         # Set Y-axis to scientific notation
         self.figure.axes[1].tick_format = '0.1e'
+
+        # Register display units for Glue unit conversion.
+        self.state.y_display_unit = flux_unit_str
+        self.state.x_display_unit = spectral_axis_unit_str
```

Comment:
This line gives me traceback when I add data to the viewer (not even the second data in the concept notebook, you already see it with the first data). I cannot figure out what triggered it. Obviously it is not a problem over at Qt glue according to https://github.com/glue-viz/glue/blob/11a5dded32a5d87409377a45729a671e83bc218a/glue/viewers/profile/qt/tests/test_data_viewer.py#L388-L389

```
----> 1 specviz.load_data(spec1, data_label="Jy_um")

File jdaviz/core/helpers.py:109, in ConfigHelper.load_data(self, data, data_label, parser_reference, **kwargs)
    107 if data_label:
    108     kwargs['data_label'] = data_label
--> 109 self.app.load_data(data, parser_reference=parser_reference, **kwargs)

File jdaviz/app.py:576, in Application.load_data(self, file_obj, parser_reference, **kwargs)
    573         parser = data_parser_registry.members.get(data_parser)
    575 if parser is not None:
--> 576     parser(self, file_obj, **kwargs)
    577 else:
    578     self._application_handler.load_data(file_obj)

File jdaviz/configs/specviz/plugins/parsers.py:121, in specviz_spectrum1d_parser(app, data, data_label, format, show_in_viewer, concat_by_file)
    118                 dfnuallorig.append(spec.uncertainty[wlind].array)
    120         elif i == 0:
--> 121             app.add_data_to_viewer(spectrum_viewer_reference_name, data_label[i])
    123 if concat_by_file and isinstance(data, SpectrumList):
    124     # If >1 spectra in the list were opened from the same FITS file,
    125     # group them by their original FITS filenames and add their combined
    126     # 1D spectrum to the DataCollection.
    127     unique_files = group_spectra_by_filename(app.data_collection)

File jdaviz/app.py:1098, in Application.add_data_to_viewer(self, viewer_reference, data_label, visible, clear_other_data)
   1095 if viewer_item is None:
   1096     raise ValueError(f"Could not identify viewer with reference {viewer_reference}")
-> 1098 self.set_data_visibility(viewer_item, data_label, visible=visible, replace=clear_other_data)

File jdaviz/app.py:1488, in Application.set_data_visibility(self, viewer_reference, data_label, visible, replace)
   1485 active_data = self.data_collection[viewer_data_labels[0]]
   1486 if (hasattr(active_data, "_preferred_translation")
   1487         and active_data._preferred_translation is not None):
-> 1488     self._set_plot_axes_labels(viewer_id)
   1490 if self.config == 'imviz':
   1491     viewer.on_limits_change()

File jdaviz/app.py:1111, in Application._set_plot_axes_labels(self, viewer_id)
   1101 """
   1102 Sets the plot axes labels to be the units of the data to be loaded.
   1103 
   (...)
   1107     The UUID associated with the desired viewer item.
   1108 """
   1109 viewer = self._viewer_by_id(viewer_id)
-> 1111 viewer.set_plot_axes()

File jdaviz/configs/specviz/plugins/viewers.py:503, in SpecvizProfileView.set_plot_axes(self)
    501 # Register display units for Glue unit conversion.
    502 self.state.y_display_unit = flux_unit_str
--> 503 self.state.x_display_unit = spectral_axis_unit_str

File echo/core.py:261, in HasCallbackProperties.__setattr__(self, attribute, value)
    260 def __setattr__(self, attribute, value):
--> 261     super(HasCallbackProperties, self).__setattr__(attribute, value)
    262     if self.is_callback_property(attribute):
    263         self._notify_global(**{attribute: value})

File jdaviz/core/freezable_state.py:16, in FreezableState.__setattr__(self, k, v)
     14 def __setattr__(self, k, v):
     15     if k[0] == '_' or k not in self._frozen_state:
---> 16         super().__setattr__(k, v)
     17     elif getattr(self, k) is None:
     18         # still allow Nones to be updated to initial values
     19         super().__setattr__(k, v)

File echo/selection.py:37, in SelectionCallbackProperty.__set__(self, instance, value)
     35 if self.comparison_type == 'equality' or (self.comparison_type is None and np.isscalar(value)):
     36     if not any(value == x for x in choices):
---> 37         raise ValueError('value {0} is not in valid choices: {1}'.format(value, choices))
     38 if self.comparison_type == 'identity' or (self.comparison_type is None and not np.isscalar(value)):
     39     if not any(value is x for x in choices):

ValueError: value um is not in valid choices: ['']
```

Reply:
Somehow `self.state.x_display_unit` is not going through the converter. I don't understand why.

Reply:
PaddedSpectrumWCS?

Reply:
@pllim - this was a bug in glue-astronomy, can you try with https://github.com/glue-viz/glue-astronomy/pull/87?

Reply:
@astrofrog , thanks! The error is gone with dev glue-astronomy, but I still cannot get it to work properly, please see https://github.com/glue-viz/glue/pull/2296/files#r1123283337